torch/distributed/_tensor/dispatch.py

```diff
@@ -190,6 +190,27 @@ def default_tensor(spec: DTensorSpec) -> torch.Tensor:
                 ):
                     local_results = op_call(*local_tensor_args, **op_info.local_kwargs)
             else:
+                # local_tensor_args need to be modified for new factory ops, potentially
```

Comment:
I think we should not do this in dispatch.py, instead we should do this in the sharding propagation, the reason is that adjusting to the local shape looks like a resharding (that's actually what we already did too) operation where we should simply generate a redistribute_schema that involves the input args that are adjusted already to local_shape. what we need to do:

1. maintain a dict of ops that map to which arg need to be adjusted to local (shape/stride) in the Propagator
2. after we get the output OpStrategy, we examine if the op is in the dict of ops, if it is, we adjust the corresponding args to local shape/stride when generating a redistribute_schema
3. return this as a redistribute_schema then dispatch.py would handle this automatically